src/Components/test/testassets/Components.WasmMinimal/Routes.razor

```diff
@@ -1,6 +1,29 @@
 ﻿@using Microsoft.AspNetCore.Components.Routing
 
-<Router AppAssembly="@typeof(Program).Assembly">
+@using Components.WasmMinimal.Pages
+@inject NavigationManager NavigationManager
+
+@code {
+    private Type? NotFoundPageType { get; set; }
+
+    protected override void OnInitialized()
+    {
+        var uri = NavigationManager.ToAbsoluteUri(NavigationManager.Uri);
+        var query = System.Web.HttpUtility.ParseQueryString(uri.Query);
+        var useCustomNotFoundPage = query["useCustomNotFoundPage"];
```

Comment:
nit: You can use `[SupplyParameterFromQuery]` public bool UseCustomNotFoundPage {get; set;}